src/types/chart/chart.ts

```diff
@@ -103,6 +107,7 @@ export type ExcelChartType = "line" | "bar" | "pie" | "combo" | "scatter";
 
 export interface ExcelChartDefinition {
   readonly title?: TitleDesign;
+  readonly titleType?: ChartTitleType;
```

Comment:
Maybe we can find a way to group the titleType and the titleDesign in one single object :) It will avoid having another field in the chart definition (and the contextCreation) that are already really long.

Reply:
```ts

export interface WaterfallChartDefinition {
  readonly title: ChartTitleConfig;
  // other properties
}

export interface ChartTitleConfig extends TitleDesign {
  readonly type?: "string" | "reference";
}

/**
 * The AbstractChartTitle interface is primarily used within the AbstractChart
 * to handle the various types of title.
 */
export interface AbstractChartTitle extends Omit<ChartTitleConfig, "text"> {
  readonly title: ChartTitle;
}

export type ChartTitle =
  | { type: "reference"; reference: Range | undefined }
  | { type: "string"; value: string };

```

Do you suggest having a structure like this?

Grouping the design in one single object and the type in another might introduce unnecessary complexity, especially considering that we're already nesting one object in AbstractChart to manage types for the title properly with TypeScript errors.

Reply:
I was thinking of something like
```js
export type ChartTitle = {
design: TitleDesign;
} & (
{type: "reference"; reference: Range | undefined}
| {type: "string"; value: string }
);

So that we will have a title object containing the design (where we can remove the `text` part) as well as the type and the reference/value :) But if it's too complex for the structure we can keep it as is

Reply:
Hi 🙂
One last question before I start making changes: Do you mean to remove `text` directly from TitleDesign, or only when using it for this ChartTitle?

Reply:
Hello :) For me, it can be fully removed for TitleDesign, as long as we use ChartTitle in the definition instead :)